### PR TITLE
feat(synthetics): CloudWatch Synthetics NodeJS Pupeteer 6.0 Runtime with NodeJS 18

### LINKED ISSUE #27337: (aws-synthetics-alpha): Node 16 End Of Life
### Describe the feature

Opeing new issue (duplicate of previously [closed issue](https://github.com/aws/aws-cdk/issues/27028)) now that `syn-nodejs-puppeteer-6.0` has been released.

Node 16 is [EOL](https://nodejs.org/en/blog/announcements/nodejs16-eol). Cloudwatch Synthetics should be able to utilise the next supported runtime (node 18), ideally before the EOL.

### Use Case

Cloudwatch Synthetic Lambda's should have the ability to use the Node 18 runtime... so that AWS clients can keep up to date with the later NodeJS software releases and stay on a supported version.

### Proposed Solution

Create a new SYNTHETICS_NODEJS runtime running on the newly configured syn-nodejs-puppeteer v6.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.83.0

### Environment details (OS name and version, etc.)

n/a

### PR DESCRIPTION
**Pull request by Abdu03**
Added `syn-nodejs-puppeteer-6.0` version to avaiable synthetic runtimes.
This version runs on Node 18 allowing users to move away from node 16 (which is now at EOL).

Changed an exisiting test to refrence the latest runtime and also updated the examples in the readme to refrence the latest runtime.

Closes #27337.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by peterwoodworth — CHANGES_REQUESTED** [review]
Hey @Abdu03, according to the build logs, the build is failing because the indentation isn't correct in `runtime.ts` on lines 125 and 138

Regarding integration tests, we'll need to modify and run the tests found in [this folder](https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk-testing/framework-integ/test/aws-synthetics/test).

**Review by mrgrain — DISMISSED** [review]
🎉🎉

File: packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts [hunk 1]
```diff
@@ -122,6 +122,21 @@ export class Runtime {
...
+   * - Chromium version 111.0.5563.146
+   *
+   * New Features:
+   * - **Dependency upgrade**: The Node.js dependency is upgraded to 18.x.
+   * Bug fixes:
+   * - **Bug fix**: Clean up core dump generated when Chromium crashes during a canary run.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Library_nodejs_puppeteer.html#CloudWatch_Synthetics_runtimeversion-nodejs-puppeteer-6.0
+   */
+   public static readonly SYNTHETICS_NODEJS_PUPPETEER_6_0 = new Runtime('syn-nodejs-puppeteer-6.0', RuntimeFamily.NODEJS);
```
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts:138]:
```suggestion
  public static readonly SYNTHETICS_NODEJS_PUPPETEER_6_0 = new Runtime('syn-nodejs-puppeteer-6.0', RuntimeFamily.NODEJS);
```

File: packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts [hunk 2]
```diff
@@ -122,6 +122,21 @@ export class Runtime {
    */
   public static readonly SYNTHETICS_NODEJS_PUPPETEER_5_1 = new Runtime('syn-nodejs-puppeteer-5.1', RuntimeFamily.NODEJS);
 
+   /**
```
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts:125]:
```suggestion
  /**
```

**Review by mrgrain — APPROVED** [review]

**Comment by Abdu03** [comment]
thank you for clarifiying @peterwoodworth 🙌 and thank you for sorting it out @mrgrain 🚀

**Pull request merged**